src/main/kotlin/com/coxautodev/graphql/tools/TypeClassMatcher.kt

```diff
@@ -25,36 +20,17 @@ internal class TypeClassMatcher(private val definitionsByName: Map<String, TypeD
         return if(potentialMatch.batched) {
             match(stripBatchedType(potentialMatch)) // stripBatchedType sets 'batched' to false
         } else {
-            match(potentialMatch, potentialMatch.graphQLType, potentialMatch.javaType, true)
+            match(potentialMatch, potentialMatch.graphQLType, potentialMatch.javaType)
         }
     }
 
-    private fun match(potentialMatch: PotentialMatch, graphQLType: GraphQLLangType, javaType: JavaType, root: Boolean = false): Match {
+    private fun match(potentialMatch: PotentialMatch, graphQLType: GraphQLLangType, javaType: JavaType): Match {
 
-        var realType = potentialMatch.generic.unwrapGenericType(javaType)
-        var optional = false
-
-        // Handle jdk8 Optionals
-        if(realType is ParameterizedType && potentialMatch.generic.isTypeAssignableFromRawClass(realType, Optional::class.java)) {
-            optional = true
-
-            if(potentialMatch.location == Location.RETURN_TYPE && !root) {
-                throw error(potentialMatch, "${Optional::class.java.name} can only be used at the top level of a return type")
-            }
-
-            realType = potentialMatch.generic.unwrapGenericType(realType.actualTypeArguments.first())
-
-            if(realType is ParameterizedType && potentialMatch.generic.isTypeAssignableFromRawClass(realType, Optional::class.java)) {
-                throw error(potentialMatch, "${Optional::class.java.name} cannot be nested within itself")
-            }
-        }
```

Comment:
I'm not sure I'm right here, but I think the above code isn't necessary anymore, as `Optional` is now handled as a `GenericWrapper`, see [`Builder#genericWrappers` initialization](https://github.com/graphql-java/graphql-java-tools/pull/120/files#diff-b2aa02d18c13c1243d9da9b0fc205a6aR235).